osquery/sql/virtual_sqlite_table.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/sql/sqlite_util.h"
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+
+Status genSqliteQueryRow(sqlite3_stmt* stmt,
+                         QueryData& qd,
+                         const fs::path& sqlite_db) {
+  Row r;
+  for (auto i{0}; i < sqlite3_column_count(stmt); ++i) {
+    auto column_name = std::string(sqlite3_column_name(stmt, i));
+    auto column_type = sqlite3_column_type(stmt, i);
+    switch (column_type) {
+    case SQLITE_TEXT: {
+      auto text_value = sqlite3_column_text(stmt, i);
+      if (text_value != nullptr) {
+        r[column_name] = std::string(reinterpret_cast<const char*>(text_value));
```

Comment:
are you sure there's truly no way to avoid this `reinterpret_cast`?

Reply:
If we want it a const char* then I think so but it's type without the cast is `const unsigned char *` which may be sufficient.

Reply:
Turns out it is not sufficient and a reinterpret cast is needed.